src/Microsoft.TestPlatform.ObjectModel/TestCase.cs

```diff
@@ -131,7 +213,17 @@ public string DisplayName
         {
             get
             {
-                return string.IsNullOrEmpty(this.displayName) ? this.FullyQualifiedName : this.displayName;
+                if(string.IsNullOrEmpty(this.displayName))
+                {
+                    if(string.IsNullOrWhiteSpace(managedType) || string.IsNullOrWhiteSpace(managedMethod))
```

Comment:
Consider adding a helper method `HasManagedMethodAndType` or some such that can be used in all of the places that need this check.